hack/release-tool.py

```diff
@@ -0,0 +1,129 @@
+                        logger.warning('You must specify "--execute" to permanently apply these changes')
+                        exit(0)
+
+                update_annotations(tag)
+                tag.replace()
+                logger.info(f'Release {release} updated successfully')
+                logger.info(f'Backup written to: {backup_filename}')
+
+        except (ValueError, OpenShiftPythonException, Exception) as e:
+            logger.error(f'Unable to update release: "{release}"')
+            raise e
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Manually accept or reject release payloads')
+    parser.add_argument('-m', '--message', help='Specifies a custom message to include with the update', default=None)
+    parser.add_argument('-r', '--reason', help='Specifies a custom reason to include with the update', default=None)
+    parser.add_argument('--execute', help='Specify to persist changes on the cluster', action='store_true')
+    parser.add_argument('--admin', help='Perform operations as "system:admin"', action='store_true')
+
+    config_group = parser.add_argument_group('Configuration Options')
+    config_group.add_argument('-v', '--verbose', help='Enable verbose output', action='store_true')
+
+    ocp_group = parser.add_argument_group('Openshift Configuration Options')
+    ocp_group.add_argument('-c', '--context', help='The OC context to use (default is "app.ci")', default='app.ci')
+    ocp_group.add_argument('-n', '--namespace', help='The namespace prefix to use (default is "ocp")', default='ocp')
+    ocp_group.add_argument('-i', '--imagestream', help='The name of the release imagestream to use (default is "release")', default='release')
+    ocp_group.add_argument('-a', '--architecture', help='The architecture of the release to process (default is "amd64")', choices=SUPPORTED_ARCHITECTURES, default='amd64')
+    ocp_group.add_argument('-p', '--private', help='Enable updates of "private" releases', action='store_true')
+
+    subparsers = parser.add_subparsers(title='subcommands', description='valid subcommands', help='Supported operations', required=True)
+    accept_parser = subparsers.add_parser('accept', help='Accepts the specified release')
+    accept_parser.set_defaults(action='accept')
+    reject_parser = subparsers.add_parser('reject', help='Rejects the specified release')
+    reject_parser.set_defaults(action='reject')
+
+    parser.add_argument('release', help='The name of the release to process (i.e. 4.10.0-0.ci-2021-12-17-144800)')
+
+    args = vars(parser.parse_args())
+
+    if args['verbose']:
+        logger.setLevel(logging.DEBUG)
+
+    context = {"context": args['context']}
+
+    if args['admin']:
+        context['as'] = 'system:admin'
+
+    validate_server_connection(context)
+    namespace, imagestream = generate_resource_values(args['namespace'], args['imagestream'], args['architecture'], args['private'])
```

Comment:
Right - you return the full namespace here, so what's being passed in as `--namespace` is not really the namespace.